server/src/main/java/org/elasticsearch/index/mapper/KeywordFieldMapper.java

```diff
@@ -897,14 +886,17 @@ private void indexValue(DocumentParserContext context, String value) {
             return;
         }
 
+        final String name = fieldType().name();
         if (value.length() > ignoreAbove) {
-            context.addIgnoredField(name());
+            context.addIgnoredField(name);
             return;
         }
 
-        value = normalizeValue(fieldType().normalizer(), name(), value);
+        if (normalizer != null) {
```

Comment:
It seems like we are optimizing either:

1. A constant vs null check
2. A method call that gets a final field.
3. The method call to normalizeValue when normalizer is keyword.

I am not sure either are important. Can you elaborate on the intention here? I fear such optimizations will easily fall off.